admin/class-acc_user_importer-admin.php

```diff
@@ -358,148 +374,192 @@ private function proccess_user_data ( $users ) {
 			else {
-				$api_response['log'] .= "&nbsp;&gt; <b>error</b>: no email given, cannot create new user account.<br/>";
-				$this->log_local_output(" > error: no email given, cannot create new user account.");
-				$update_this_user = false;
+				$this->log_dual(" > error: no email given, cannot create new user account.");
 				$update_errors[] = $user['FirstName'] . " " . $user['LastName'] . " (" . $user_contact_id . ")";
 			}
 			
+			//If user already exists, update only if something changed.
+			//Do a bunch of compare to decide if an update is needed.
+			//This logic needs to be undated each time we add a new user field.
+			if ($update_this_user && $user_already_exists) {
 
-			//only run if indicated
-			if ($update_this_user) {
+				$update_this_user = false;
 
-				$wp_user = wp_insert_user( $user_data ) ;
-				
-				//attempt to add user
-				if ( is_wp_error($wp_user) ) {
-					
-					$api_response['log'] .= "&nbsp;&gt; failed to update user<br/>";
-					$api_response['log'] .= "&nbsp;&gt; WP:" . $wp_user->get_error_message() . "<br/>";
-					$this->log_local_output(" > failed to update user");
-					$this->log_local_output(" > WP:" . $wp_user->get_error_message());
+				// Get the user object
+				$user_meta = get_userdata($user_id);
+
+				// Get all the user roles as an array.
+				$user_roles = $user_meta->roles;
+				// Check if user has expired or ex-member role. If so, update role to default
+				if ( in_array( $role_expiry1, $user_roles, true ) ) {
+					$this->log_dual(" > User is " . $role_expiry1 . ", refreshing role to " . $default_role);
+					$update_this_user = true;
+					$user_data["role"] = $default_role;
+					$role_refreshed[] = $user_contact_id;
+					$role_refreshed_email[] = $user_email;
+				} elseif (in_array( $role_expiry2, $user_roles, true ) ) {
+					$this->log_dual(" > User is " . $role_expiry2 . ", refreshing role to " . $default_role);
+					$update_this_user = true;
+					$user_data["role"] = $default_role;
+					$role_refreshed[] = $user_contact_id;
+					$role_refreshed_email[] = $user_email;
+				}
+
+				//Check if email changed
+				if (!$update_this_user) {
+					if ($user_email != $user_meta->user_email) {
+						$this->log_dual(" > email changed from " . $user_meta->user_email . " to " . $user_email);
+						$update_this_user = true;
+					}
+				}
+
+				//Check if HomePhone changed
+				if (!$update_this_user) {
+					if ($user["HomePhone"] != $user_meta->home_phone) {
+						$this->log_dual(" > home phone changed from " . $user_meta->home_phone . " to " . $user["HomePhone"]);
+						$update_this_user = true;
+					}
+				}
+
+				//Check if Cell phone changed
+				if (!$update_this_user) {
+					if ($user["Cell Phone"] != $user_meta->cell_phone) {
+						$this->log_dual(" > cell phone changed from " . $user_meta->cell_phone . " to " . $user["Cell Phone"]);
+						$update_this_user = true;
+					}
+				}
+
+				//Check if MEMBERSHIP_N changed
+				if (!$update_this_user) {
```

Comment:
I decided to remove the above if.  This way the logic is simpler and the log will indicate all fields that changed for a given user, not only the first one.